web/packages/teleterm/src/ui/DocumentCluster/UnifiedResources.tsx

```diff
@@ -0,0 +1,287 @@
+    sort: { fieldName: 'name', dir: 'ASC' },
+  });
+
+  const { documentsService, rootClusterUri } = useWorkspaceContext();
+  const loggedInUser = useWorkspaceLoggedInUser();
+  const { canUse: hasPermissionsForConnectMyComputer, agentCompatibility } =
+    useConnectMyComputerContext();
+
+  const isRootCluster = props.clusterUri === rootClusterUri;
+  const canAddResources = isRootCluster && loggedInUser?.acl?.tokens.create;
+
+  const canUseConnectMyComputer =
+    isRootCluster &&
+    hasPermissionsForConnectMyComputer &&
+    agentCompatibility === 'compatible';
+
+  const { fetch, resources, attempt } = useUnifiedResourcesFetch({
+    fetchFunc: useCallback(
+      async (paginationParams, signal) => {
+        const response = await retryWithRelogin(
+          appContext,
+          props.clusterUri,
+          () =>
+            appContext.resourcesService.listUnifiedResources(
+              {
+                clusterUri: props.clusterUri,
+                searchAsRoles: false,
+                sortBy: {
+                  isDesc: params.sort.dir === 'DESC',
+                  field: params.sort.fieldName,
+                },
+                search: params.search,
+                kindsList: params.kinds,
+                query: params.query,
+                pinnedOnly: params.pinnedOnly,
+                startKey: paginationParams.startKey,
+                limit: paginationParams.limit,
+              },
+              signal
+            )
+        );
+
+        return {
+          startKey: response.nextKey,
+          agents: response.resources,
+          totalCount: response.resources.length,
+        };
+      },
+      [
+        appContext,
+        params.kinds,
+        params.pinnedOnly,
+        params.query,
+        params.search,
+        params.sort.dir,
+        params.sort.fieldName,
+        props.clusterUri,
+      ]
+    ),
+  });
+
+  useEffect(() => {
+    const { cleanup } = onResourcesRefreshRequest(() =>
+      fetch({
+        force: true,
+        fromStart: true,
+      })
+    );
+    return cleanup;
+  }, [onResourcesRefreshRequest, fetch]);
+
+  return (
+    <SharedUnifiedResources
+      params={params}
+      setParams={setParams}
+      updateUnifiedResourcesPreferences={() => alert('Not implemented')}
+      onLabelClick={() => alert('Not implemented')}
+      pinning={{ kind: 'hidden' }}
+      resources={resources
+        .map(resource => {
+          switch (resource.kind) {
+            case 'server': {
+              const { resource: server } = resource;
+              const serverId = routing.parseServerUri(server.uri)?.params
+                ?.serverId;
+              if (!serverId) {
+                return;
+              }
+              return {
+                resource: {
+                  kind: 'node' as const,
+                  labels: server.labelsList,
+                  id: serverId,
+                  name: server.name,
```

Comment:
Isn't server ID equivalent to its name?

https://github.com/gravitational/teleport/blob/b5696c059e82157336c4b4b2f62f22549fbcba1a/lib/teleterm/clusters/cluster_servers.go#L89

https://github.com/gravitational/teleport/blob/b5696c059e82157336c4b4b2f62f22549fbcba1a/lib/teleterm/apiserver/handler/handler_servers.go#L66-L68

Reply:
Thanks, I forgot about it!